### PR TITLE
Refactor config reading

### PR DESCRIPTION
**Pull request by cry**
Subclass the config object from Flask and merge in our envvar changes into the default object.

### CODE REVIEW

**Comment by cry** [comment]
Do not merge, needs more changes

File: flask_core/config.py [hunk 1]
```diff
@@ -62,21 +70,35 @@ def __init__(self, **kwargs):
...
-            )
-        except KeyError as e:
-            raise RuntimeError(f"Required config option {e} not set and not provided from environment.")
+        self.FLAG_IDS = getattr(self, "FLAG_IDS", None) or os.environ.get("FLAG_IDS", None)
+        if self.FLAG_IDS is not None:
+            self.FLAG_IDS = self.FLAG_IDS.split(",")
+
+        self.FLAG_WRAP = getattr(self, "FLAG_WRAP", None) or os.environ.get("FLAG_WRAP", None)
+        self.FLAG_SECRET = getattr(self, "FLAG_SECRET", None) or os.environ.get("FLAG_SECRET", None)
+        self.DB_CONNECTION_STRING = getattr(self, "DB_CONNECTION_STRING", None) or os.environ.get(
```
Comment (zainafzal08) [inline flask_core/config.py:90]: Can we even build an app with no database without the entire system shitting itself?

File: flask_core/config.py [hunk 2]
```diff
@@ -62,21 +70,35 @@ def __init__(self, **kwargs):
...
+            "DB_CONNECTION_STRING", None
+        )
+
+    def validate(self):
+        required = ["FLAG_IDS", "FLAG_WRAP", "FLAG_SECRET", "DB_CONNECTION_STRING"]
```
Comment (zainafzal08) [inline flask_core/config.py]: in future would be nice to make DB connection optional, not every site needs a db

**Review by zainafzal08 — APPROVED** [review]

File: flask_core/config.py (hunk 2 above)
Comment (cry) [reply flask_core/config.py]: We technically don't need to make it a required parameter, I'll make it optional (or you can)

**Pull request merged**